File: solutions/csharp/squeaky-clean/SqueakyClean.cs
using System;
using System.Text;

public static class Identifier
{
    public static string Clean(string identifier)
    {
        bool isLowerCaseGreek(char c) => c >= 'α' && c <= 'ω';
        bool kebabFound = false;
        var result = new StringBuilder();
        foreach (char c in identifier)
        {
            result.Append(c switch
            {
                // Replace any spaces encountered with underscores
                _ when char.IsWhiteSpace(c) => '_',
                // Replace control characters with the upper case string "CTRL"
                _ when char.IsControl(c) => "CTRL",
                // Convert kebab-case to camelCase
                _ when kebabFound => char.ToUpper(c),
                // Omit Greek lower case letters
                _ when isLowerCaseGreek(c) => default,
                // Omit characters that are not letters
                _ when !char.IsLetter(c) => default,
                _ => c,
            });
            kebabFound = c.Equals('-');
        }
        return result.ToString();
    }
}

File: solutions/java/squeaky-clean/src/main/java/SqueakyClean.java
class SqueakyClean {
    static String clean(String identifier) {
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < identifier.length(); i++) {
            char c = identifier.charAt(i);

            if (Character.isSpaceChar(c)) {
                // Replace any spaces encountered with underscores
                result.append('_');

            } else if (c == '-') {
                // Convert kebab-case to camelCase
                i += 1;
                result.append(Character.toUpperCase(identifier.charAt(i)));

            } else if (Character.isDigit(c)) {
                // Convert leetspeak to normal text
                switch (c) {
                    case '4':
                        result.append('a');
                        break;
                    case '3':
                        result.append('e');
                        break;
                    case '0':
                        result.append('o');
                        break;
                    case '1':
                        result.append('l');
                        break;
                    case '7':
                        result.append('t');
                        break;
                    default:
                        result.append(c);
                        break;
                }
            } else if (Character.isLetter(c)) {
                // Omit characters that are not letters
                result.append(c);
            }
        }

        return result.toString();
    }
}
